### ISSUE TITLE
non-service-account icommands are noisy when hosts_config.json not readable

### ISSUE THREAD

**Issue by trel** [irods/irods#5625]
### iRODS Version, OS and Version

upgraded 4.2.8

### What did you try to do?

When the permissions on `/etc/irods/hosts_config.json` are not readable by others...

```
$ sudo ls -l /etc/irods/hosts_config.json
-rw------- 1 irods irods 91 May 10 21:55 /etc/irods/hosts_config.json
```

From non-service-account... any iCommand prints this error...
```
$ ils
 ERROR: resolve_hostname_from_hosts_config :: caught exception: 
/tempZone/home/rods:
$ imkdir new
 ERROR: resolve_hostname_from_hosts_config :: caught exception: 
$ icd new
 ERROR: resolve_hostname_from_hosts_config :: caught exception: 
$
```

but all still have return code `0` (which is correct/good).

### Expected behavior

Less noise if/when `hosts_config.json` is unreadable?   Maybe this is working as expected? 

### Other information

This was on a 4.1.x -> 4.2.8 upgraded system.   Perhaps the permissions need to be better-preserved during upgrade?

```
-rw------- 1 irods irods   107 May 10 21:55 host_access_control_config.json
-rw-rw-r-- 1 irods irods    29 May 10 21:55 host_access_control_config.json.prev.1620698107.87
-rw------- 1 irods irods    91 May 10 21:55 hosts_config.json
-rw-rw-r-- 1 irods irods    27 May 10 21:55 hosts_config.json.prev.1620698107.87
-rw------- 1 irods irods  3249 May 10 21:55 server_config.json
-rw-rw-r-- 1 irods irods  1789 May 10 21:55 server_config.json.prev.1620698107.87
```

**Comment by cookie33** [comment]
We have the same errors after an upgrade of iRODS 4.1.11 to 4.2.8.

```
$ ils -l
 ERROR: resolve_hostname_from_hosts_config :: caught exception:
/frank/home/robertv:
  robertv           0 eudat;eudatCache           12 2019-12-20.15:59 & test_data.txt
  robertv           1 eudat;eudatPnfs           12 2019-12-20.16:00 & test_data.txt
  ```

Are there plans to fix this?

What are the required permissions of the files in `/etc/irods` under linux? Otherwise we could set it ourself after an upgrade.

**Comment by trel** [comment]
Is this the same as https://github.com/irods/irods/issues/5447?

**Comment by cookie33** [comment]
The error output is the same. But I had:
```
/etc/irods
rods$ ls -l
total 188
...
-rw------- 1 rods rods   667 Aug 11 15:32 hosts_config.json
-rw-rw-r-- 1 rods rods   438 Aug 11 15:32 hosts_config.json.prev.1628688737.16
...
```
Changed the protection:
````
/etc/irods
rods$ chmod 664 hosts_config.json
```
And now the protection was ok:
```
/etc/irods
rods$ ls -l
total 188
...
-rw-rw-r-- 1 rods rods   667 Aug 11 15:32 hosts_config.json
-rw-rw-r-- 1 rods rods   438 Aug 11 15:32 hosts_config.json.prev.1628688737.16
```

And now the command as the non-irods admin user functions ok:
```
$ ils -l
/frank/home/robertv:
  robertv           0 eudat;eudatCache           12 2019-12-20.15:59 & test_data.txt
  robertv           1 eudat;eudatPnfs           12 2019-12-20.16:00 & test_data.txt
```

**Comment by korydraughn** [comment]
This issue was resolved in 4.2.9 by the following PR: https://github.com/irods/irods/pull/5403

Resolution of hostnames via the hosts_config.json file was reimplemented and better protects server-side code from clients such as the icommands.

**Comment by trel** [comment]
marking as resolved and closing.
